pengine/native.c

```diff
@@ -2092,6 +2092,37 @@ native_expand(resource_t * rsc, pe_working_set_t * data_set)
     }
 }
 
+#define is_runnable(a) ((a) && (a)->node && \
+    (((a)->flags & (pe_action_optional|pe_action_runnable|pe_action_pseudo)) == pe_action_runnable))
```

Comment:
Note that we already have the tail encoded in `action_is_pending` in `tools/crm_resource_runtime.c`.
Would it deserve some form of code refactoring and shrinking?
It would then also make the line shorter :)